alternator/server.cc

```diff
@@ -424,11 +426,15 @@ future<executor::request_return_type> server::handle_api_request(std::unique_ptr
     }
     _pending_requests.enter();
     auto leave = defer([this] () noexcept { _pending_requests.leave(); });
-    //FIXME: Client state can provide more context, e.g. client's endpoint address
-    // We use unique_ptr because client_state cannot be moved or copied
-    executor::client_state client_state = username.empty()
-        ? service::client_state{service::client_state::internal_tag()}
-        : service::client_state{service::client_state::internal_tag(), _auth_service, _sl_controller, username};
+    // we are required to set timeout_config even though it isn't really
+    // relevant for Alternator, which defines its own timeouts explicitly.
+    // FIXME: create the timeout_config object only once when creating the server.
+    updateable_timeout_config timeout_config(_proxy.data_dictionary().get_config());
```

Comment:
This should be a member of server, since it's updateable. It doesn't need to be queried again and again.

Reply:
Ah, you even have a comment for it. Why not fix the FIXME?

Reply:
When I did this part of, I wanted to make progress quickly and not spend yet more time on making this a tiny bit more efficient. But I guess I can/should do it now.

Reply:
Done. I added a comment before the new class member explaining it is *not* the real timeout configuration of Alternator (yuck).